core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -343,8 +356,15 @@ RefreshingAddressResolverGroup build(EventLoopGroup eventLoopGroup) {
             if (dnsServerAddressStreamProvider != null) {
                 builder.nameServerProvider(dnsServerAddressStreamProvider);
             }
-            if (dnsQueryLifecycleObserverFactory != null) {
-                builder.dnsQueryLifecycleObserverFactory(dnsQueryLifecycleObserverFactory);
+            if (dnsQueryLifecycleObserverFactory == null) {
+                builder.dnsQueryLifecycleObserverFactory(new DefaultDnsQueryLifecycleObserverFactory(
+                        meterRegistry, new MeterIdPrefix("armeria.client.dns.queries")));
```

Comment:
Let's add `assert meterRegistry != null` so that IDE does not complain.
Let's exctrat
```java
new DefaultDnsQueryLifecycleObserverFactory(
    meterRegistry, new MeterIdPrefix("armeria.client.dns.queries"))
```
before line 359

Reply:
@minwoox Interesting observation. If I extract the construction of DefaultDnsQueryLifecycleObserverFactory and then pass the same DefaultDnsQueryLifecycleObserverFactory to both cases then the UT break because we are now using the same DnsQueryLifecycleObserverFactory. The UT break because the write counts are 2 times. Question is we should use the same DnsQueryLifecycleObserverFactory and not create it independently in both cases. I think this might be a bug. What do you think? The breaking UT when I extract the creation of dnsQueryLifecycleObserverFactory is below - 
```
io.micrometer.core.instrument.MeterRegistryjava.lang.String 
Expecting map:
 <{"armeria.client.connections.lifespan#count{protocol=h2c}"=0.0, "armeria.client.connections.lifespan#max{protocol=h2c}"=0.0, "armeria.client.connections.lifespan#total{protocol=h2c}"=0.0, "armeria.client.dns.queries#count{cause=none,name=foo.com.,result=success}"=2.0, "armeria.client.dns.queries.written#count{name=foo.com.,server=0:0:0:0:0:0:0:1}"=2.0}>
to contain:
 <[MapEntry[key="armeria.client.dns.queries.written#count{name=foo.com.,server=0:0:0:0:0:0:0:1}", value=1.0]]>
but could not find the following map entries:
 <[MapEntry[key="armeria.client.dns.queries.written#count{name=foo.com.,server=0:0:0:0:0:0:0:1}", value=1.0]]>
```

Reply:
Had a chat with @amitvc and I hope it succeeds this time. 😄

Reply:
@minwoox the build is successful I think. https://ci.appveyor.com/project/line/armeria/builds/36414924/job/82eers20is37trhi